serverless/functions/rooms-webhook.protected.js

```diff
@@ -2,13 +2,50 @@
 
 exports.handler = async function (context, event, callback) {
   const common = require(Runtime.getAssets()['/common.js'].path);
-  const { axiosClient, response } = common(context, event, callback);
+  const { axiosClient, response, getStreamMapItem } = common(context, event, callback);
 
   const client = context.getTwilioClient();
 
   const { StatusCallbackEvent, RoomSid } = event;
 
-  if (StatusCallbackEvent === 'room-ended') {
+  if (StatusCallbackEvent === 'participant-connected') {
+    let streamSyncClient;
+
+    // Get stream sync client
+    try {
+      const streamMapItem = await getStreamMapItem(RoomSid);
+      streamSyncClient = await client.sync.services(streamMapItem.data.sync_service_sid);
+    } catch (e) {
+      console.error(e);
+      response.setStatusCode(500);
+      response.setBody({
+        error: {
+          message: 'error getting stream sync client',
+          explanation: e.message,
+        },
+      });
+      return callback(null, response);
+    }
+
+    // Remove user from viewers map
+    try {
+      await streamSyncClient.syncMaps('viewers').syncMapItems(event.ParticipantIdentity).remove();
```

Comment:
Do we also need to remove them from the raisedHands map?

Reply:
For this PR this is handled in the `join-stream-as-speaker` endpoint because that is where it existed previously and it works. But in the next PR that adds the speakers map I did move it to this webhook because it is a better time to do it.